### PR TITLE
Quickfix for path traversal issue

### PR DESCRIPTION
**Pull request by alexAubin**
## Problem

Soooo, turns out that nextcloud is vulnerable to path traveral issues ... Going to https://domain.tld/nextcloud../html/index.nginx-debian.html might show you the nginx page... Sometime it only shows a blank white page, not sure why, but even then I can access some other dummy files (so, possibly secrets stored in other apps folders...)

Since nextcloud is one of the most used app it's a bit critical to fix this issue I guess..

## Solution

Apparently, the path traversal issues requires specific ingredient to work : 
- a `location /foo` *without* the `/` at the end
- an `alias` statement to `/var/www/foo/` *with* a `/` at the end ...

Here I simply removed the `/` at the end as it seems to be the easiest solution to fix this... But I didn't test it :/ At least the discussion is opened :stuck_out_tongue_winking_eye: :sweat_smile: 

## PR Status
- [X] Code finished.
- [ ] Tested with Package_check.
- [ ] Fix or enhancement tested.
- [ ] Upgrade from last version tested.
- [ ] Can be reviewed and tested.

## Validation
---
*Minor decision*
- **Upgrade previous version** : 
- [x] **Code review** : JimboJoe
- [x] **Approval (LGTM)** : JimboJoe
- [x] **Approval (LGTM)** : Maniack C
- **CI succeeded** : 
[Build Status/badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/nextcloud_ynh%20fix-path-traversal-issue%20(Official)/)  
When the PR is marked as ready to merge, you have to wait for 3 days before really merging it.

### CODE REVIEW

**Comment by maniackcrudelis** [comment]
I've just tested https://domain.tld/nextcloud../html/index.nginx-debian.html a hour ago and it didn't work.
Just now it works !!!

That the same url !
And package check is testing `$check_domain$check_path../html/index.nginx-debian.html` which is also the same url !
There's some magic somewhere...

File: conf/nginx.conf [hunk 1]
```diff
@@ -6,7 +6,7 @@ location = /.well-known/caldav {
 }
 
 location ^~ __PATH__ {
-  alias __FINALPATH__/;
+  alias __FINALPATH__;
```
Comment (maniackcrudelis) [inline conf/nginx.conf]: I'd rather prefer the common (and working for sure) fix, https://github.com/YunoHost/example_ynh/pull/45/files than removing this slash.

Comment (alexAubin) [reply conf/nginx.conf]: Yes why not, but that will also affect the `location __PATH__ {` which happens later ... so might be less trivial ... Just removing this `/` seems be more trivial (though I haven't really checked if that does work ...)

Comment (maniackcrudelis) [reply conf/nginx.conf]: That's the point, I'm not sure it works without consequences. Otherwise we probably wouldn't have add a rewriting rule.

**Comment by alexAubin** [comment]
>There's some magic somewhere...

 A̴̬̔l̴͓̉l̶̪͝ ̷̡̆
Ć̶̼o̸̺͛m̷̱͆p̸͕͊u̵̖͂t̶̛̟e̴͈̅r̵̠̕s̴̗͛
     ̷̈͜A̷̲̾r̵̼͑e̷̘͛ ̶̨̀
  Ḇ̸́ȑ̵͖o̵͓̓ḵ̵̕ĕ̸͙n̴̜͝

File: conf/nginx.conf (hunk 1 above)
Comment (alexAubin) [reply conf/nginx.conf]: Alrighty I tested it on a test server and in fact it's not clear that it really does solve the issue ... still shows a blank page when doing path traversal.

With the `/` at the end of the location, it does show a clear 404 error which sounds better

I commited the change :wink:

Comment (alexAubin) [reply conf/nginx.conf]: (Also changed the other `location __PATH__` in the middle to `location __PATH__/` and it seems to be okay ... Didn't do a full clean check tho)

Comment (maniackcrudelis) [reply conf/nginx.conf]: Meanwhile, I try to know why package_check doesn't trig an error on nextcloud

Comment (maniackcrudelis) [reply conf/nginx.conf]: On LXC the default file has "Welcome to nginx" as title instead of "Welcome to nginx on Debian"...

Comment (alexAubin) [reply conf/nginx.conf]: (╯°□°）╯︵ ┻━┻

Comment (maniackcrudelis) [reply conf/nginx.conf]: Also, the title changed since the time we have implement this test. Of course... It would have been too easy otherwise...

**Comment by maniackcrudelis** [comment]
Looks ok

**Comment by JimboJoe** [comment]
Will be merged in 3 days.

**Comment by alexAubin** [comment]
Shall we merge it in testing after https://github.com/YunoHost-Apps/nextcloud_ynh/pull/181 ?

**Pull request merged**